src/components/CustomButton/AnimatedButton.js

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import { ActivityIndicator, TouchableWithoutFeedback, Animated } from 'react-native';
+import styles from './styles';
+
+class CustomButton extends React.Component {
+  constructor(props) {
+    super(props);
+    const { style = {} } = props;
+    this.width = new Animated.Value(style.width || 150);
```

Comment:
If you really want animate width there are two ways to do it:
1) Use [layout animation ](https://medium.com/@Jpoliachik/react-native-s-layoutanimation-is-awesome-4a4d317afd3e)
Just change your width and run animation. I.e. this.setState({ width: 0 }, () => LayoutAnimation.configureNext(LayoutAnimation.Presets.spring);)

2) Change the scale of the button and useNativeDriver:true for smooth animation
3) Use several components with different width one with opacity: 0, second with opacity: 1 and just toggle. Also useNativerDriver: true

P.s. community grows every day and now we have such amazing library like 
[https://github.com/kmagiera/react-native-reanimated](https://github.com/kmagiera/react-native-reanimated)

It's the future of animation in RN and allows you to make more flexible and interesting things in animation in RN. By the way all animation runs in native thread by default ( you don't need to specify useNativeDriver: true )